connectors/sources/graphql.py

```diff
@@ -0,0 +1,449 @@
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for GraphQL.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "http_endpoint": {
+                "label": "HTTP URL & Endpoint",
+                "order": 1,
+                "type": "str",
+            },
+            "http_method": {
+                "display": "dropdown",
+                "label": "GET/POST",
+                "options": [
+                    {"label": "GET", "value": "get"},
+                    {"label": "POST", "value": "post"},
+                ],
+                "order": 2,
+                "type": "str",
+                "value": "POST",
+            },
+            "authentication_method": {
+                "display": "dropdown",
+                "label": "Authentication Method",
+                "options": [
+                    {"label": "No Auth", "value": "none"},
+                    {"label": "Basic Auth", "value": "basic"},
+                    {"label": "Bearer Token", "value": "bearer"},
+                ],
+                "order": 3,
+                "type": "str",
+                "value": "none",
+            },
+            "username": {
+                "depends_on": [{"field": "authentication_method", "value": "basic"}],
+                "label": "Username",
+                "order": 4,
+                "type": "str",
+            },
+            "password": {
+                "depends_on": [{"field": "authentication_method", "value": "basic"}],
+                "label": "Password",
+                "order": 5,
+                "sensitive": True,
+                "type": "str",
+            },
+            "token": {
+                "depends_on": [{"field": "authentication_method", "value": "bearer"}],
+                "label": "Bearer Token",
+                "order": 6,
+                "sensitive": True,
+                "type": "str",
+            },
+            "graphql_query": {
+                "display": "textarea",
+                "label": "GraphQL Body",
+                "order": 7,
+                "type": "str",
+            },
+            "graphql_variables": {
+                "depends_on": [{"field": "http_method", "value": "post"}],
+                "display": "textarea",
+                "label": "Graphql Variables",
+                "order": 8,
+                "type": "str",
+                "required": False,
+            },
+            "graphql_object_list": {
+                "label": "GraphQL Objects List",
+                "order": 9,
+                "tooltip": "List of object names from the query that needs to be indexed.",
+                "type": "list",
+            },
+            "headers": {
+                "label": "Headers",
+                "order": 10,
+                "type": "str",
+                "required": False,
+            },
+            "pagination_enabled": {
```

Comment:
I think we should be explicit what pagination strategy we use? Should we instead have a dropdown with multiple values:
- no pagination
- offset-based pagination
- cursor-based pagination

Right now it's not clear what does `pagination enabled` refers to. Also the tooltip should have clear explanation about the variables that we expect in the query

Reply:
Also I think we should assume reasonable defaults but allow the user to configure the pagination variables for offset and curosr based pagination. In different implementations of grpahql server they can be named slightly differently (e.g. `offset` and `limit` and `skip` and `take` (keystoneJS), also they can be custom named). Should we introduce more RFCs fields (with depend_on pagaintion type relation) that allow user to define the pagination variable names used by their framework?. It should be fine to set reasonable default values e.g. offset/limit

Reply:
Currently, I am not able to find source with offset-based pagination. So, I have pushed only cursor-based pagination. Will add offset-based pagination in follow-up PR